app/workers/delete_object_hierarchy_worker.rb

```diff
@@ -106,13 +106,13 @@ def destroy_has_many_association
       main_object.public_send("#{reflection.name.to_s.singularize}_ids").each do |associated_object_id|
         associated_object = reflection.class_name.constantize.new
         associated_object.id = associated_object_id
-        delete_associated_object_later(associated_object)
+        delete_associated_object_later(associated_object) if associated_object.id
       end
     end
 
     def destroy_has_one_association
       associated_object = main_object.public_send(reflection.name)
-      delete_associated_object_later(associated_object) if associated_object
+      delete_associated_object_later(associated_object) if associated_object.try(:id)
```

Comment:
It is similar to https://github.com/3scale/porta/pull/1010/files#r304992523

I could reproduce it because in one of the tests for [#981 [credit card] unstore if gateway configured](https://github.com/3scale/porta/pull/981) I temporarily added a `dependent: :destroy` in here:
https://github.com/3scale/porta/blob/762be0132ce13f3927e3ed50e39320c92020a2fe/app/models/account/payment_details.rb#L7
And it was raising this same error for this:
https://github.com/3scale/porta/blob/762be0132ce13f3927e3ed50e39320c92020a2fe/app/models/account/payment_details.rb#L28-L46

Meaning that if it doesn't exist, it builds it, so the `associated_object` will be a `PaymentDetail` without an `ID`.


I couldn't add a test for this one because I don't know which what other model this happens, and that was just a temporary change for testing that PR.